packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -192,40 +193,66 @@ contract Deploy is Deployer {
         console.log("Deployment context: %s", deploymentContext);
     }
 
-    /// @notice Deploy all of the L1 contracts
+    /// @notice Deploy all of the L1 contracts necessary for a full Superchain with a single Op Chain.
     function run() public {
-        console.log("Deploying L1 system");
-
-        deployProxies();
-        deployImplementations();
+        console.log("Deploying a fresh OP Stack including SuperchainConfig");
 
         deploySafe();
-        transferProxyAdminOwnership(); // to the Safe
+        setupSuperchain();
+        setupOpChain();
+    }
 
-        initializeDisputeGameFactory();
-        initializeSystemConfig();
-        initializeL1StandardBridge();
-        initializeL1ERC721Bridge();
-        initializeOptimismMintableERC20Factory();
-        initializeL1CrossDomainMessenger();
-        initializeL2OutputOracle();
-        initializeOptimismPortal();
+    ////////////////////////////////////////////////////////////////
+    //           High Level Deployment Functions                  //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Deploy a full system with a new SuperchainConfig
+    ///         The Superchain system has 3 singleton contracts which lie outside of an OP Chain:
+    ///         1. The SuperchainConfig contract
+    ///         2. The ProtocolVersions contract
+    function setupSuperchain() public {
+        console.log("Setting up Superchain");
+
+        // Deploy a new ProxyAdmin and AddressManager
+        // This proxy will be used on the SuperchainConfig and ProtocolVersions contracts, as well as the contracts
+        // in the OP Chain system.
+        deployAddressManager();
+        deployProxyAdmin();
+        transferProxyAdminOwnership();
+
+        // Deploy the SuperchainConfigProxy
+        deployERC1967Proxy("SuperchainConfigProxy");
+        deploySuperchainConfig();
+        initializeSuperchainConfig();
+
+        deployERC1967Proxy("ProtocolVersionsProxy");
+        deployProtocolVersions();
         initializeProtocolVersions();
+    }
+
+    /// @notice Deploy a new OP Chain, with an existing SuperchainConfig provided
+    function setupOpChain() public {
+        console.log("Deploying OP Chain");
+
+        // Ensure that the requisite contracts are deployed
+        mustGetAddress("SuperchainConfigProxy");
+        mustGetAddress("SystemOwnerSafe");
+        mustGetAddress("AddressManager");
+        mustGetAddress("ProxyAdmin");
```

Comment:
This is an alternative to what I did on `feat/securityCouncil`, which would accept an argument for the superchainConfig address.

I think that this approach is conservative, and good enough for now. More thinking needs to be done to determine how this should be setup for independent OP chain deployments.

https://github.com/ethereum-optimism/optimism/blob/feat/security-council/packages/contracts-bedrock/scripts/Deploy.s.sol#L109-L118